packages/utils/src/lib/reports/generate-md-report.ts

```diff
@@ -26,19 +30,25 @@ export function auditDetailsAuditValue({
   )} (score: ${formatReportScore(score)})`;
 }
 
+function hasCategories(
+  report: ScoredReport,
+): report is ScoredReport & { categories: ScoredCategoryConfig[] } {
+  return !!report.categories && report.categories.length > 0;
+}
+
 export function generateMdReport(
   report: ScoredReport,
   options?: MdReportOptions,
 ): string {
   return new MarkdownDocument()
     .heading(HIERARCHY.level_1, REPORT_HEADLINE_TEXT)
-    .$if(report.categories.length > 0, doc =>
-      doc.$concat(
-        categoriesOverviewSection(report),
-        categoriesDetailsSection(report),
-      ),
+    .$concat(
+      ...(hasCategories(report)
+        ? [categoriesOverviewSection(report), categoriesDetailsSection(report)]
+        : []),
```

Comment:
Unfortunately, I had to refactor the `build-md`'s `$if` out as it only accepted a boolean condition, so it did not narrow types in the body. As a result, `categories` were still interpreted as potentially undefined inside the `$if` block, leading to TypeScript errors.